binary_and_decimal.rb

```diff
@@ -3,7 +3,17 @@
 #    appears as a binary number. Calculate the decimal value for this binary
 #    number using the algorithm you devised in class. Print out the decimal value.
 def binary_to_decimal
-  puts "Not implemented"
+  binary_num = ""
+  decimal = 0
+  8.times do |i|
+    digit = rand(2)
+    binary_num = digit.to_s + binary_num
+    decimal += 2**i * digit
```

Comment:
You have an error in decimal computation. If the binary number is 10000000 then your program will compute this to be 1 in decimal and not 128. To fix this, you'll need to account for the exponent of 2 to be reverse index. In this case `decimal += 2**(7-i) * digit` This will give you the exponent of 7 for the first binary digit (at index 0) and exponent of 0 for the last binary digit (at index 7).